### ISSUE TITLE
Auto generate promisified functions

### ISSUE THREAD

**Issue by JumpLink** [gjsify/ts-for-gir#121]
@CharlieQLe says in #106 
> This might be out of scope since it requires the usage of another function, but including promisified functions would be nice, just to avoid regenerating the types.

The automatic generation of promisified functions could be a nice feature, but should be handled in a separate PR after #106 is merged.
